ext/objspace/objspace_dump.c

```diff
@@ -37,22 +40,104 @@ struct dump_config {
     unsigned int full_heap: 1;
 };
 
-PRINTF_ARGS(static void dump_append(struct dump_config *, const char *, ...), 2, 3);
 static void
-dump_append(struct dump_config *dc, const char *format, ...)
+dump_flush(struct dump_config *dc)
 {
-    va_list vl;
+    if (dc->buffer_len) {
+        if (dc->stream) {
+            fwrite(dc->buffer, 1, dc->buffer_len, dc->stream);
+        }
+        else if (dc->string) {
+            rb_str_cat(dc->string, dc->buffer, dc->buffer_len);
+        }
+        dc->buffer_len = 0;
+    }
+}
+
+static inline void
+buffer_inc_capa(struct dump_config *dc, unsigned long requested)
+{
+    unsigned long required;
+
+    if (requested + dc->buffer_len >= dc->buffer_capa) {
+        dump_flush(dc);
+    }
+
+    for (required = dc->buffer_capa; requested > required - dc->buffer_len; required <<= 1);
+
+    if (required > dc->buffer_capa) {
+        REALLOC_N(dc->buffer, char, required);
```

Comment:
`REALLOC_N` can trigger GC inside.  Is that okay at this particular moment?

Reply:
That's a good point. I don't know wether it's GC safe here. However looking at it again, since I added some code to flush the buffer before increasing the buffer, this is pretty much dead code. I think I'll refactor a bit and use a static size buffer.

Reply:
The buffer is now statically allocated on the heap. But I think triggering GC is safe, or at least it can already be triggered today by other parts of `dump_all`.